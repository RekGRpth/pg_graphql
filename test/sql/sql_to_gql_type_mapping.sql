/*
Monitor if the SQL -> GQL type map changes
*/
select
    pt.oid::regtype,
    graphql.sql_type_to_graphql_type(pt.oid::regtype) as graphql_type
from
    pg_type pt
    join pg_namespace pn
        on pt.typnamespace = pn.oid
where
    pt.typname not like '\_%'
    and graphql.sql_type_to_graphql_type(pt.oid::regtype) <> 'String'
    or pt.typname similar to '(text|varchar|char)'
order by
    graphql.sql_type_to_graphql_type(pt.oid::regtype),
    pt.typname
