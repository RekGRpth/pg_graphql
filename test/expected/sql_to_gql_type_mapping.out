/*
Monitor if the SQL -> GQL type map changes
*/
select
    pt.oid::regtype,
    graphql.sql_type_to_graphql_type(pt.oid::regtype) as graphql_type
from
    pg_type pt
    join pg_namespace pn
        on pt.typnamespace = pn.oid
where
    pt.typname not like '\_%'
    and graphql.sql_type_to_graphql_type(pt.oid::regtype) <> 'String'
    or pt.typname similar to '(text|varchar|char)'
order by
    graphql.sql_type_to_graphql_type(pt.oid::regtype),
    pt.typname
             oid             | graphql_type 
-----------------------------+--------------
 bigint                      | BigInt
 boolean                     | Boolean
 date                        | Date
 timestamp without time zone | Datetime
 timestamp with time zone    | Datetime
 real                        | Float
 double precision            | Float
 smallint                    | Int
 integer                     | Int
 json                        | JSON
 jsonb                       | JSON
 "char"                      | String
 text                        | String
 character varying           | String
 time without time zone      | Time
 uuid                        | UUID
(16 rows)

